cdap-common/src/test/java/io/cdap/cdap/common/http/AuthenticationChannelHandlerTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.http;
+
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.proto.security.Credential;
+import io.cdap.cdap.security.spi.authentication.UnauthenticatedException;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class AuthenticationChannelHandlerTest {
+
+  private final DefaultHttpRequest req = mock(DefaultHttpRequest.class, RETURNS_DEEP_STUBS);
+  private AuthenticationChannelHandler handler;
+  private ChannelHandlerContext ctx;
+
+  @Before
+  public void initHandler() {
+    boolean internalAuthEnabled = true;
+    handler = new AuthenticationChannelHandler(internalAuthEnabled);
+    ctx = mock(ChannelHandlerContext.class, RETURNS_DEEP_STUBS);
+  }
+
+  @Test(expected = UnauthenticatedException.class)
+  public void testEmptyHeaderThrowsUnauthenticatedException() throws Exception {
+    when(req.headers().get(Constants.Security.Headers.RUNTIME_TOKEN)).thenReturn("");
```

Comment:
nit: this is a bit brittle since we're directly hooking the method which can be susceptible to any slight change to the header read logic. In the future, it might be a bit more robust to just create a request with a set of headers.

Reply:
Fixed